src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1293,63 +1294,80 @@ learnPointsTo ::
+                    , [ "Tried to read something of size:"
+                      , show (Crucible.bytesToInteger sz)
+                      ]
+                    , [ "And type:", show (Crucible.ppMemType memTy) ]
                     ]
-                  , bullets '-' (map Crucible.ppSomeAlloc possibleAllocs)
-                  ]
-               -- This information tends to be overwhelming, but might be useful?
-               -- We should brainstorm about better ways of presenting it.
-               -- PP.<$$> PP.text (unwords [ "Here are the details on why reading"
-               --                          , "from each matching write failed"
-               --                          ])
-               -- PP.<$$> PP.text (show err)
+            case res of
+              Crucible.NoErr pred_ res_val -> do
+                pred_' <- case maybe_cond of
+                  Just cond -> do
+                    cond' <- instantiateExtResolveSAWPred sc cc (ttTerm cond)
+                    liftIO $ W4.impliesPred sym cond' pred_
+                  Nothing -> return pred_
+                addAssert pred_' $ Crucible.SimError loc $
+                  Crucible.AssertFailureSimError (show $ PP.vcat $ summarizeBadLoad) ""
+                pure Nothing <* matchArg opts sc cc spec prepost res_val memTy val'
+              _ -> do
+                -- When we have a concrete failure, we do a little more computation to
+                -- try and find out why.
+                let (blk, _offset) = Crucible.llvmPointerView ptr1
+                pure $ Just $ PP.vcat . (summarizeBadLoad ++) $
+                  case W4.asNat blk of
+                    Nothing -> [PP.text "<Read from unknown allocation>"]
+                    Just blk' ->
+                      let possibleAllocs =
+                            Crucible.possibleAllocs blk' (Crucible.memImplHeap mem)
+                      in [ PP.text $ unwords
+                           [ "Found"
+                           , show (length possibleAllocs)
+                           , "possibly matching allocation(s):"
+                           ]
+                         , bullets '-' (map Crucible.ppSomeAlloc possibleAllocs)
+                         ]
+                      -- This information tends to be overwhelming, but might be useful?
+                      -- We should brainstorm about better ways of presenting it.
+                      -- PP.<$$> PP.text (unwords [ "Here are the details on why reading"
+                      --                          , "from each matching write failed"
+                      --                          ])
+                      -- PP.<$$> PP.text (show err)
+       SymbolicSizeValue expected_arr_tm expected_sz_tm ->
+         do maybe_allocation_array <- liftIO $
+              Crucible.asMemAllocationArrayStore sym Crucible.PtrWidth ptr1 (Crucible.memImplHeap mem)
+            liftIO $ putStrLn $ "***" ++ show maybe_allocation_array
```

Comment:
Just a debugging message? We should probably leave this out or properly integrate it into the logging system.

Reply:
ups, thanks!